content/raw/chat/username_avatar.md

```diff
@@ -0,0 +1,132 @@
+---
+title: "username & profile spec"
+description: ""
+date: 2020-12-06T08:48:57+00:00
+lastmod: 2020-12-06T08:48:57+00:00
+draft: false
+images: []
+menu:
+  raw:
+    parent: "chat"
+weight: 100
+toc: true
+---
+
+# Username & Profile Spec
+
+## Motivation
+
+Users need a mechanism to identity themselves to other users that is easy to use and at the same time somewhat resistant to impersonation attacks. At the time of writing Status uses deterministic 3 word usernames and ENS usernames. Unfortunately 3 word usernames have shown to be quite confusing for new users as they are used to set their own usernames in any chat app.
+Here we introduce support for users to set their own display name and introduce at least 3 mechaniscs to minimize the risk of impersonation attacks. The first mechanism is the use of deterministic sequence of emojis to uniquely identity a profile, the second mechanism is the use of a deterministic sequence of colors around the profile to differentiate it from other profiles which effectively replaces the 3 word usernames and identicons with a similar deterministic identifier, the third mechanism involves preventing duplicate displaynames from joining the same community. There is also an identity verification mechanism that is used to ensure that the user is who they say they are described in the Identity Verifcation Spec.
+
+## Definitions
+
+- 3 word usernames - A 3 word username is a deterministic sequence of words that uniquely identifies a profile. see [spec](https://specs.status.im/spec/2#3-word-pseudonym--whisperwaku-key-fingerprint)
```

Comment:
After evaluation, we determined that those 3 word are not unique, rather, they cover about 3 bytes of the pubkey, therefore leaving a lot of margin for conflicts.